presto-main/src/main/java/com/facebook/presto/util/MergeSortedPages.java

```diff
@@ -132,10 +134,12 @@ private MergeSortedPages() {}
         });
     }
 
-    private static WorkProcessor<PageWithPosition> pageWithPositions(WorkProcessor<Page> pages, LocalMemoryContext memoryContext)
+    private static WorkProcessor<PageWithPosition> pageWithPositions(WorkProcessor<Page> pages, AggregatedMemoryContext aggregatedMemoryContext)
     {
         return pages.flatMap(page -> {
+            LocalMemoryContext memoryContext = aggregatedMemoryContext.newLocalMemoryContext();
             memoryContext.setBytes(page.getRetainedSizeInBytes());
```

Comment:
Is this needed? We usually don't account individual pages.
If `flatMap` guarantees that computations are not parallel/interleaved, then we could drop memory accounting here.

Reply:
> If flatMap guarantees that computations are not parallel/interleaved, then we could drop memory accounting here.

Computations can be yielded or blocked so the page might be kept in memory. PageBuilder can also be full. In this case (in spill context) one can open multiple of files (hunderds) each with large page, so we rather need to account memory for that.

> Is this needed? We usually don't account individual pages.

Often we do. Example: `MergingPageOutput`, `PageProcessor`, Especially in spill context we want to be careful not to underaccount (at least I remember such reasoning when spill for aggregation was landing)